File: leetcode/01.defang-ip-address/defang.go
import ("strings")

// Golang version of "defanging" and IP address
// Replaces "." with "[.]"
// "255.100.50.0" becomes "255[.]100[.]50[.]0"
//
func defangIPaddr(address string) string {
    return strings.Replace(address, ".", "[.]", -1)
}

File: leetcode/01.defang-ip-address/defang.java
/**
/* Java tool for "defanging" an IP address.
/* Replaces "." with "[.]"
/* "255.100.50.0" becomes "255[.]100[.]50[.]0"
*/
class Solution {
    public String defangIPaddr(String address) {
        StringBuffer defanged = new StringBuffer();
        for (int i = 0; i < address.length(); i++) {
            char c = address.charAt(i);

            if (c == '.') {
                defanged.append('[');
                defanged.append(c);
                defanged.append(']');
            } else {
                defanged.append(c);
            }
        }
        return defanged.toString();
    }
}